ffi/diplomat/src/timezone_formatter.rs

```diff
@@ -0,0 +1,115 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::errors::ffi::ICU4XError;
+    use crate::locale::ffi::ICU4XLocale;
+    use crate::provider::ffi::ICU4XDataProvider;
+    use crate::timezone::ffi::ICU4XCustomTimeZone;
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatResult;
+    use icu_datetime::time_zone::FallbackFormat;
+    use icu_datetime::time_zone::IsoFormat;
+    use icu_datetime::time_zone::IsoMinutes;
+    use icu_datetime::time_zone::IsoSeconds;
+    use icu_datetime::TimeZoneFormatter;
+
+    #[diplomat::opaque]
+    /// An ICU4X TimeZoneFormatter object capable of formatting an [`ICU4XCustomTimeZone`] type (and others) as a string
+    #[diplomat::rust_link(icu::datetime::TimeZoneFormatter, Struct)]
+    pub struct ICU4XTimeZoneFormatter(pub TimeZoneFormatter);
+
+    #[diplomat::enum_convert(IsoFormat, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::time_zone::IsoFormat, Enum)]
+    pub enum ICU4XIsoTimeZoneFormat {
+        Basic,
+        Extended,
+        UtcBasic,
+        UtcExtended,
+    }
+
+    #[diplomat::enum_convert(IsoMinutes, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::time_zone::IsoMinutes, Enum)]
+    pub enum ICU4XIsoTimeZoneMinuteDisplay {
+        Required,
+        Optional,
+    }
+
+    #[diplomat::enum_convert(IsoSeconds, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::time_zone::IsoSeconds, Enum)]
+    pub enum ICU4XIsoTimeZoneSecondDisplay {
+        Optional,
+        Never,
+    }
+
+    impl ICU4XTimeZoneFormatter {
+        /// Creates a new [`ICU4XTimeZoneFormatter`] from locale data.
+        ///
+        /// Uses localized GMT as the fallback format.
+        #[diplomat::rust_link(icu::datetime::TimeZoneFormatter::try_new_unstable, FnInStruct)]
+        #[diplomat::rust_link(icu::datetime::time_zone::FallbackFormat, Enum, compact)]
+        pub fn try_new_with_localized_gmt_fallback(
```

Comment:
thought: was hoping we could avoid this, but i guess this is the best way to do it